src/components/attendee/ScheduleOverlay/ScheduleEvent.tsx

```diff
@@ -0,0 +1,72 @@
+import { Link } from "react-router-dom";
+import { faBookmark as regularBookmark } from "@fortawesome/free-regular-svg-icons";
+import { faBookmark as solidBookmark } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { differenceInCalendarDays } from "date-fns";
+
+import { ATTENDEE_INSIDE_URL, STRING_DASH_SPACE, STRING_SPACE } from "settings";
+
+import { ScheduledEvent } from "types/venues";
+
+import { eventEndTime, eventStartTime, isEventLive } from "utils/event";
+import { formatDateRelativeToNow, formatTimeLocalised } from "utils/time";
+import { generateUrl } from "utils/url";
+
+import { useSpaceById } from "hooks/spaces/useSpaceById";
+import { useWorldById } from "hooks/worlds/useWorldById";
+
+import CN from "./ScheduleOverlay.module.scss";
+
+type ScheduleEventProps = {
+  event: ScheduledEvent;
+  bookmarkEvent: (event: ScheduledEvent) => void;
+};
+
+export const ScheduleEvent: React.FC<ScheduleEventProps> = ({
+  event,
+  bookmarkEvent,
+}) => {
+  const isCurrentEventLive = isEventLive(event);
+  const { world } = useWorldById(event.worldId);
+  const { space } = useSpaceById(event.spaceId);
```

Comment:
[NOTE]

Would have been nicer if these both accepted options object so they can be used as 

```suggestion
  const { world } = useWorldById(event);
  const { space } = useSpaceById(event);
```

But, I don't see use of either `isLoading` and `isLoaded`, so I guess that limits the use to only `generateUrl()`

Reply:
[OOS] 
On a refactore, the above as well as returning extra info like `worldId` and `spaceSlug` so entire object wouldn't figure in the destruction might help re-renders.

[NOTE] to self: `useMemo()` inside those hooks

Reply:
done [8f7d0bc](https://github.com/sparkletown/sparkle/pull/2964/commits/8f7d0bcb22aad901b7d6456cb7b36d6787fe4ac1)